sys/cheri/cheric.h

```diff
@@ -208,6 +201,22 @@ cheri_bytes_remaining(const void * __capability cap)
 
 #endif	/* __has_feature(capabilities) */
 
+#if defined(_KERNEL) && defined(__CHERI_PURE_CAPABILITY__)
```

Comment:
Maybe this should go into a kernel-only header?

Reply:
Will probably do this first in the purecap kernel branch as I need to fix includes.